### PR TITLE
[AUTOGENERATED] rocm7.1_internal_testing_IFU_2025-09-09

### PR DESCRIPTION
**Pull request by pragupta**
rocm_base: 681e60e175c

Tested this PR on MI300x using `registry-sc-harbor.amd.com/framework/compute-rocm-dkms-no-npi-hipclang:16623_ubuntu24.04_py3.12_pytorch_rocm7.1_internal_testing_681e60e1`

Ran the following UTs:
test_nn, test_torch, test_cuda, test_ops, test_unary_ufuncs, test_autograd, inductor/test_torchinductor

All ran fine, attaching logs!
[default_ut.log](https://github.com/user-attachments/files/22244687/default_ut.log)

Successful wheel build job with this branch: http://rocm-ci.amd.com/view/preview/job/pytorch2.8-manylinux-wheels-preview/116/

### CODE REVIEW

**Review by pruthvistony — APPROVED** [review]
Rubber-stamping the PR. Build is successful and conflicts have been resolved.

**Comment by pragupta** [comment]
Do not merge yet, vllm team is testing with this branch and if perf looks good then we will merge it. Otherwise, wait till 9/15.

File: .ci/docker/ci_commit_pins/triton.txt [hunk 1]
```diff
@@ -1,5 +1 @@
-<<<<<<< HEAD
 56765e8c1f6490e21312b46242ed78cb2dd46d35
```
Comment (pruthvistony) [inline .ci/docker/ci_commit_pins/triton.txt]: It will be updated to new branch

File: aten/src/ATen/cuda/detail/OffsetCalculator.cuh [hunk 2]
```diff
@@ -49,20 +49,12 @@ struct OffsetCalculator {
 #if defined(USE_ROCM)
     if ((dims > 0) && (dims <= 2)) {
       auto divmod = sizes_[0].divmod(linear_idx);
-<<<<<<< HEAD
-      #pragma unroll
```
Comment (pruthvistony) [inline aten/src/ATen/cuda/detail/OffsetCalculator.cuh:52]: if we merge this line, then next IFU this will not show as conflict.

File: aten/src/ATen/cuda/tunable/GemmHipblaslt.h [hunk 3]
```diff
@@ -91,10 +91,6 @@ constexpr hipDataType HipDataTypeFor<c10::Float4_e2m1fn_x2>() {
 #if ROCM_VERSION >= 70000
   return HIP_R_4F_E2M1;
 #else
-<<<<<<< HEAD
-  // Return HIP_R_4F_E2M1 enum value for earlier ROCm version.
```
Comment (pruthvistony) [inline aten/src/ATen/cuda/tunable/GemmHipblaslt.h:94]: Same comment as above. 
In current scenario better to merge <<HEAD block, to avoid conflicts at next IFU.

File: aten/src/ATen/native/Convolution.cpp [hunk 4]
```diff
@@ -1406,15 +1406,8 @@ static inline at::MemoryFormat determine_backend_memory_format(
     case ConvBackend::Miopen:
     case ConvBackend::MiopenDepthwise:
     case ConvBackend::MiopenTranspose:
-<<<<<<< HEAD
```
Comment (pruthvistony) [inline aten/src/ATen/native/Convolution.cpp]: Matching upstream, it is fine.

File: aten/src/ATen/native/ConvUtils.h [hunk 5]
```diff
@@ -362,18 +362,12 @@ inline at::MemoryFormat miopen_conv_suggest_memory_format(const at::Tensor& inpu
   }
 
   // TODO: Remove PYTORCH_MIOPEN_SUGGEST_NHWC once ROCm officially supports NHWC in MIOpen
-<<<<<<< HEAD
```
Comment (pruthvistony) [inline aten/src/ATen/native/ConvUtils.h]: Matching upstream, it is fine.

File: aten/src/ATen/native/miopen/Conv_miopen.cpp [hunk 6]
```diff
@@ -765,244 +765,7 @@ void miopen_convolution_add_bias_(CheckedFrom c, const TensorArg& output, const
   */
 }
 
-<<<<<<< HEAD
-// see NOTE [ Convolution design ] in src/Aten/native/cudnn/Conv.cpp
```
Comment (pruthvistony) [inline aten/src/ATen/native/miopen/Conv_miopen.cpp:693]: Please confirm with Dmitry if all this code can it be removed.

cc @dnikolaev-amd

**Review by pruthvistony — COMMENTED** [review]
On NHWC batchnorm, need Dmitry confirmation.

Other conflicts are resolved properly. AFAIK.

File: .ci/docker/ci_commit_pins/triton.txt (hunk 1 above)
Comment (pragupta) [reply .ci/docker/ci_commit_pins/triton.txt]: Yes, this will be handled separately

File: aten/src/ATen/cuda/detail/OffsetCalculator.cuh (hunk 2 above)
Comment (pragupta) [reply aten/src/ATen/cuda/detail/OffsetCalculator.cuh:52]: Sorry, I don't follow. I chose upstream change so next time we do IFU it won't show as merge conflict. If I choose HEAD, then we are picking local changes and may show as a merge conflict.

File: aten/src/ATen/native/miopen/Conv_miopen.cpp (hunk 6 above)
Comment (dnikolaev-amd) [reply aten/src/ATen/native/miopen/Conv_miopen.cpp:693]: @jeffdaily could you take a look?

**Review by jithunnair-amd — APPROVED** [review]
LGTM. Hope other IFUs get shorter, both in time and diffs :)
